api/src/main/java/com/velocitypowered/api/plugin/Plugin.java

```diff
@@ -0,0 +1,52 @@
+package com.velocitypowered.api.plugin;
+
+import java.lang.annotation.ElementType;
+import java.lang.annotation.Retention;
+import java.lang.annotation.RetentionPolicy;
+import java.lang.annotation.Target;
+import java.util.regex.Pattern;
+
+/**
+ * Annotation used to describe a Velocity plugin.
+ */
+@Retention(RetentionPolicy.RUNTIME)
+@Target(ElementType.TYPE)
+public @interface Plugin {
+    /**
+     * The pattern plugin IDs must match. Plugin IDs may only contain
+     * alphanumeric characters, dashes or underscores, must start with
+     * an alphabetic character and cannot be longer than 64 characters.
+     */
+    Pattern ID_PATTERN = PluginDescription.ID_PATTERN;
```

Comment:
This is too much of an internal implementation detail. We should just document the ID requirement.